### ISSUE TITLE
Single argument separated by '&' has an ugly line break for typescript parser

### ISSUE THREAD

**Issue by faheem-akhtar** [prettier/prettier#6988]
**Prettier 1.19.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6VACAZgVymGAS2gEMAbDSAWwAdoEYMB3QmACwxwHMyBPDMoShwARgCc4JANYAdYQA86YxpCgBnRgFleAGWhcAciSpwAJgCVJBAGJ4CxKOQDCEWvVgBGJBlv4i0FzdhWAAeAEE9KENjMwBJWDgxLBIwOA8MADIMCP0jE1N4+CSUuAAmAD4MAF4MAAoaMQgaNQBKasrgAF85OXRsO39HCmo6YMYWdgwAN3JCUwEhUQlpOThFCGVKaA0MbUjo-MsUmF97UjJA0YZS71PBy-cYcP28uITi1PSAH2yXmIL3slUhVqnUGk1Wu0MF0elA+rg-A5yFsggxmKwODNBPNBMJxJJZAolCptlpdLl-kcbAMkRdXFdYABmW40gL0x7PCn5QqJIFpAA0vy5byKfJBNXqjWabSqHU6IH5ICagzUyFAJDEjSYAAUNQhVSgSFMIHMFSBxCkpHAYABlGgpIRcZAwMQ4OCKtgwKhkADqbFYcDU9tSNv1rEIU1YvGQ4DUqsVQjUiRg2rEJC4VBIyGSZCTioAVmp5AAhNNgK22mI6RbZ8h5kCF+Q2x1kOAARRwEHgtdz7pA9rESbEMZgvBogbAYkINBgZoaQhgPrm7GQAA4AAyK8FJn1pmgxhqBxJTOBmgCOnfgqYhMZIagAtMIzGYzRIL4QJKn05me-Wk1RCGdV0+zUFt20vU8kBzesYBIEQl1MFckFKRUXRIQhcS4QIfxQKB6DNHAkwAFTgg1oL7KY3XiUwGBtSdpxgMIoFMG1R1bX84E6TogA)
```sh
--parser typescript
```

**Input:**
```tsx
// functional component with ugly linebreak
export const MyLongNamedReactFunctionalComponent1: FunctionComponent<ALongNamedInterface1 & ALongNamedInterface2> = (props) => {}

// functional component with valid linebreak
export const MyLongNamedReactFunctionalComponent2: FunctionComponent<ALongNamedInterface1 | ALongNamedInterface2> = (props) => {}

// functional component with valid linebreak
export const MyLongNamedReactFunctionalComponent3: FunctionComponent<ALongNamedInterface1, ALongNamedInterface2> = (props) => {}
```

**Output:**
```tsx
// functional component with ugly linebreak
export const MyLongNamedReactFunctionalComponent1: FunctionComponent<ALongNamedInterface1 &
  ALongNamedInterface2> = props => {};

// functional component with valid linebreak
export const MyLongNamedReactFunctionalComponent2: FunctionComponent<
  ALongNamedInterface1 | ALongNamedInterface2
> = props => {};

// functional component with valid linebreak
export const MyLongNamedReactFunctionalComponent3: FunctionComponent<
  ALongNamedInterface1,
  ALongNamedInterface2
> = props => {};

```

**Expected behavior:**
The first component in the above code should be parsed like the other two.

**Comment by thorn0** [comment]
Keywords to make this issue easier to find: intersection, intersections

Related discussions: https://github.com/prettier/prettier/pull/3988

**Comment by thorn0** [comment]
fixed in 2.0, needs tests
